File: modules/BBWM.SystemSettings/ProjectSettings.cs
using BBWM.FileStorage;

namespace BBWM.SystemSettings;

/// <summary>
/// Represents site appearance settings.
/// </summary>
public class ProjectSettings : IMutableSystemConfigurationSettings
{
    public static readonly string DefaultName = "Blueberry Web Template v3";
    public static readonly string DefaultTheme = "ultima-indigo-compact";
    public static readonly string DefaultLogoIconUrl = "favicon.ico";
    public static readonly string DefaultLogoImageUrl = "/assets/images/logo.png";

    private string _theme;

    public string Name { get; set; } = DefaultName;

    public string ThemeCode
    {
        get => string.IsNullOrEmpty(_theme) ? DefaultTheme : _theme;
        set => _theme = value;
    }

    public int? LogoImageId { get; set; }

    public int? LogoIconId { get; set; }
}

public class ProjectSettingsImages
{
    public FileDetailsDTO LogoIcon { get; set; }
    public FileDetailsDTO LogoImage { get; set; }
}

File: project/BBWT.Client/src/app/main/system-configuration/classes/project-settings.ts
import { SystemConfigurationResolveData } from "../system-configuration-resolve-data";
import { SettingsSectionsName } from "../settings-sections-name";
import { FileDetails } from "../../file-storage";

export class ProjectSettings {
    static readonly DefaultName = "Blueberry Web Template v3";
    static readonly DefaultThemeCode = "ultima-indigo";
    static readonly DefaultLogoIconUrl = "favicon.ico";
    static readonly DefaultLogoImageUrl = "/assets/images/logo.png";

    name: string;
    themeCode: string;
    logoImageId: number;
    logoIconId: number;

    static parse(data: SystemConfigurationResolveData) {
        return data.getConfigSection<ProjectSettings>(SettingsSectionsName.ProjectSettings);
    }
}

export class ProjectSettingsImages {
    logoIcon: FileDetails;
    logoImage: FileDetails;
}